src/api/app/models/project.rb

```diff
@@ -31,6 +31,7 @@ class ForbiddenError < APIException
   end
 
   before_destroy :cleanup_before_destroy
+  after_destroy :delete_on_backend
```

Comment:
Maybe [after_destroy_commit](http://guides.rubyonrails.org/active_record_callbacks.html#transaction-callbacks) would be even better